### PR TITLE
Add volume for nginx-gate for read-only mode

### PR DESCRIPTION
**Pull request by bhushanthakur93**
Related APPSRE-9925

Based on the discussion in https://github.com/app-sre/nginx-gate/pull/10#discussion_r1609163979

### CODE REVIEW

File: openshift/app-interface.yaml [hunk 1]
```diff
@@ -66,6 +66,11 @@ objects:
                       - app-interface
                   topologyKey: topology.kubernetes.io/zone
                 weight: 100
+        volumes:
+        - name: var-cache-nginx
+          emptyDir: { }
```
Comment (rporres) [inline openshift/app-interface.yaml:71]: shouldn't this cache go to memory?

Comment (hemslo) [reply openshift/app-interface.yaml:71]: We can try `tmpfs` and bump memory request/limit based on size

https://kubernetes.io/docs/concepts/storage/volumes/#emptydir

```yaml
emptyDir:
  medium: Memory
```

Comment (rporres) [reply openshift/app-interface.yaml:71]: Yes, that was exactly what I had in mind.

**Review by hemslo — APPROVED** [review]
LGTM!

**Review by rporres — APPROVED** [review]

**Pull request merged**